storm-client/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1505,35 +1508,36 @@ public static StormTopology addVersions(StormTopology topology) {
         Map<String, Object> defaultsConf = null;
         Map<String, Object> stormConf = null;
         for (String part: cp) {
-            File f = new File(part);
-            if (f.isDirectory()) {
-                if (defaultsConf == null) {
-                    defaultsConf = readConfIgnoreNotFound(yaml, new File(f, "defaults.yaml"));
-                }
-                
-                if (stormConf == null) {
-                    stormConf = readConfIgnoreNotFound(yaml, new File(f, "storm.yaml"));
+            String wildcardSuffix = File.separator + "*";
+            if (part.endsWith(wildcardSuffix)) {
+                // wildcard is given
+                // in java classpath, '*' is translated to '*.jar'
+                File dir = new File(part.substring(0, part.length() - wildcardSuffix.length()));
```

Comment:
Could we use Path for creating paths instead of manipulating strings? (asking because it's nice if the code is obviously portable)

Reply:
I just get rid of manipulating string via File's method. Please take a look at new commit.

Reply:
Thanks, looks good